tests/src/rules/no-duplicates.js

```diff
@@ -129,6 +129,30 @@ ruleTester.run('no-duplicates', rule, {
       errors: ['\'./foo\' imported multiple times.', '\'./foo\' imported multiple times.'],
     }),
 
+    // #2347: duplicate identifiers should be removed
+    test({
+      code: "import {a} from './foo'; import { a } from './foo'",
+      output: "import {a} from './foo'; ",
+      errors: ['\'./foo\' imported multiple times.', '\'./foo\' imported multiple times.'],
+      parser: parsers.BABEL_OLD, // Needed to avoid "A fatal parsing error occurred: Parsing error: Identifier 'a' has already been declared"
+    }),
+
+    // #2347: duplicate identifiers should be removed
+    test({
+      code: "import {a,b} from './foo'; import { b, c } from './foo'; import {b,c,d} from './foo'",
+      output: "import {a,b, c ,d} from './foo';  ",
+      errors: ['\'./foo\' imported multiple times.', '\'./foo\' imported multiple times.', '\'./foo\' imported multiple times.'],
+      parser: parsers.BABEL_OLD, // Needed to avoid "A fatal parsing error occurred: Parsing error: Identifier 'b' has already been declared"
+    }),
+
+    // #2347: duplicate identifiers should be removed, but not if they are adjacent to comments
+    test({
+      code: "import {a} from './foo'; import { a/*,b*/ } from './foo'",
+      output: "import {a, a/*,b*/ } from './foo'; ",
+      errors: ['\'./foo\' imported multiple times.', '\'./foo\' imported multiple times.'],
+      parser: parsers.BABEL_OLD, // Needed to avoid "A fatal parsing error occurred: Parsing error: Identifier 'a' has already been declared"
+    }),
```

Comment:
I added one more test case.
This feels a bit icky to me.
I could:

(A) leave it as-is
(B) treat this case as a "problematic comment" and bail (don't attempt to auto-fix)
(C) change the split logic for identifiers so it doesn't blindly split on commas and it takes comments into account -- this could get pretty complex, I guess I'd need a regex

What do you think?

Reply:
It is almost always the proper choice to refuse to autofix any nodes adjacent to comments, because "comment attachment" (whether it points to the previous or the next node) is not a solved problem.

Reply:
Okay, so are you saying (A) the current behavior is acceptable?

What I meant by (B) was to change the logic inside `hasProblematicComments`:

```diff
diff --git a/src/rules/no-duplicates.js b/src/rules/no-duplicates.js
index bd094cb..c84a314 100644
--- a/src/rules/no-duplicates.js
+++ b/src/rules/no-duplicates.js
@@ -223,7 +223,7 @@ function hasProblematicComments(node, sourceCode) {
   return (
     hasCommentBefore(node, sourceCode) ||
     hasCommentAfter(node, sourceCode) ||
-    hasCommentInsideNonSpecifiers(node, sourceCode)
+    hasCommentInside(node, sourceCode)
   );
 }
 
@@ -241,18 +241,10 @@ function hasCommentAfter(node, sourceCode) {
     .some(comment => comment.loc.start.line === node.loc.end.line);
 }
 
-// Checks whether `node` has any comments _inside,_ except inside the `{...}`
-// part (if any).
-function hasCommentInsideNonSpecifiers(node, sourceCode) {
+// Checks whether `node` has any comments _inside_.
+function hasCommentInside(node, sourceCode) {
   const tokens = sourceCode.getTokens(node);
-  const openBraceIndex = tokens.findIndex(token => isPunctuator(token, '{'));
-  const closeBraceIndex = tokens.findIndex(token => isPunctuator(token, '}'));
-  // Slice away the first token, since we're no looking for comments _before_
-  // `node` (only inside). If there's a `{...}` part, look for comments before
-  // the `{`, but not before the `}` (hence the `+1`s).
-  const someTokens = openBraceIndex >= 0 && closeBraceIndex >= 0
-    ? tokens.slice(1, openBraceIndex + 1).concat(tokens.slice(closeBraceIndex + 1))
-    : tokens.slice(1);
+  const someTokens = tokens.slice(1);
   return someTokens.some(token => sourceCode.getCommentsBefore(token).length > 0);
 }
 ```

If you'd prefer I do this, I'm happy to do it, but three existing tests are failing and I'd need to change those too.